armada_jupyter/__main__.py

```diff
@@ -8,8 +14,40 @@ def submit(file: str):
     """
     Creates new JupyterLab pods defined in the submission file
     """
+    submit_worker(file, ArmadaClient)
+
+
+def submit_worker(file: str, armada_client: ArmadaClient):
+    """
+    A worker function for submit command.
+
+    This function is separated from submit command to allow for easier testing.
+    """
+
+    typer.echo(f"Getting Submission Objects from {file}")
+    submission = submissions.convert_to_submission(file)
+
+    typer.echo(f"Submitting {len(submission.jobs)} Jobs to Armada")
+
+    for job in submission.jobs:
+
+        client, job_id = armada.submit(submission, job, armada_client)
+        typer.echo(f"Submitted Job {job_id} to Armada")
+
+        # Sleep to make sure that job-set-id is created
+        time.sleep(3)
```

Comment:
This is how the API was designed - to the best of my knowlegde working with the python client over summer there is no other option

Reply:
I agree with @Sharpz7.  This is a problem we found with the Armada API and their focus on being "eventually consistent".  

If we this proves to be brittle, we could add an environment variable for this sleep behavior or remove the logic around checking if pod is ready.  

As I write this, we may want to relax this.  Since Armada is a queueing solution, you could actually have days/weeks before your job gets queued.  Right now we assume that it takes 3 seconds for your job to be queued and running but this is an invalid assumption.

Reply:
I think I'd want more discussion with GR internally before I make that kind of call.

Reply:
Just to clarify, this sleep is waiting until the `event stream` becomes active, not for the job to start running - I think those are independent. Therefore, this sleep time is pretty consistently under 3 seconds?

Maybe polling is acceptable in this case?

Reply:
I have done what I said above, with three "tries" to connect to the event stream, 1s between each try. Regardless of the job states, the job should generally always be created within this time.

Reply:
>As I write this, we may want to relax this. Since Armada is a queueing solution, you could actually have days/weeks before >your job starts running. Right now we assume that it takes 3 seconds for your job to be queued and running but this is an >invalid assumption.

Based on this I am going to add a config option to toggle whether the user wants to use the event stream to wait or not